### PR TITLE
New Addon Markdown

### PR DESCRIPTION
**Pull request by annando**
This addon parses user input for Markdown content as well. This is highly experimental and could possibly lead to wrong parsing of BBCode elements. But since it can be enabled and disabled on a user level, users can test it out and report problems.

### CODE REVIEW

**Comment by Ravenbird** [comment]
Wow thank you! :-) Now I just have to see how I build it into my Friendica instance.

**Comment by Ravenbird** [comment]
So ich habe jetzt mal ein wenig mit dem Markdown AddOn herum gespielt und bisher sieht es sehr gut aus. Ich habe aber noch keine komplexeren Sachen wie z. B. Tabellen etc. getestet. Werde ich jetzt nach und nach machen. :-) DANKE!

**Comment by Ravenbird** [comment]
So I've been playing around with the Markdown AddOn and so far it looks very good. But I haven't tested more complex things like tables etc. yet. I will do this bit by bit now. :-) THANK YOU!

File: markdown/README [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Markdown
+========
+
+The Markdown addon parses user input for new items and comments via the Markdown parser.
+This enables users to use the Markdown syntax additionally to the BBCode syntax.
```
Comment (tobiasd) [inline markdown/README]: I like the irony. Adding a Markdown addon, but not using Markdown for the `README.md` but instead plain text :-D

Comment (annando) [reply markdown/README]: That's the result of "copy and paste".

File: markdown/markdown.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	if (!local_user()) {
+		return;
+	}
+
+	$enabled = intval(PConfig::get(local_user(), 'markdown', 'enabled'));
+
+	$t = Renderer::getMarkupTemplate('settings.tpl', 'addon/markdown/');
+	$s .= Renderer::replaceMacros($t, [
+		'$title'   => L10n::t('Markdown'),
+		'$enabled' => ['enabled', L10n::t('Enable Markdown parsing'), $enabled, L10n::t('If enabled, self created items will additionally be parsed via Markdown.')],
```
Comment (tobiasd) [inline markdown/markdown.php]: Could you please extract the strings to be translated for the addon?

**Comment by Ravenbird** [comment]
A short question. Do you know what dialect of markdown this is? Github?

**Comment by annando** [comment]
The uses Markdown parser is the one that we are using for the Diaspora integration.

File: markdown/markdown.php (hunk 2 above)
Comment (annando) [reply markdown/markdown.php]: Done

File: markdown/markdown.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+	PConfig::set(local_user(), 'markdown', 'enabled', intval($_POST['enabled']));
+}
+
+function markdown_post_local_start(App $a, &$request) {
+	if (empty($request['body']) || !PConfig::get(local_user(), 'markdown', 'enabled')) {
+		return;
+	}
+
+	$request['body'] = Markdown::toBBCode($request['body']);
```
Comment (MrPetovan) [inline markdown/markdown.php]: Isn’t this going to parse Markdown code is `[code]` tags?

Comment (AlfredSK) [reply markdown/markdown.php]: This works as expected. The markdown stuff is posted as plain text.

``` 
[code]### Headline
* markdown test[/code]
```

But this doesnt work...

``` 
[code]
### Headline
* markdown test
[/code]
``` 

The result is totally weird:

``` 
[h3]Headline[/h3][list][*]markdown test
[/list]
``` 
(that's the output in the post. :-/ )

Comment (MrPetovan) [reply markdown/markdown.php]: So it doesn't work as expected.

Comment (annando) [reply markdown/markdown.php]: We surely have to solve this. Like I said:
> This is highly experimental

We also have to see if there are other BBCode elements that need to be escaped as well.

**Pull request merged**